test/legacy_test/test_batch_norm_op_v3.py

```diff
@@ -0,0 +1,524 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import numpy as np
+from eager_op_test import (
+    OpTest,
+    _set_use_system_allocator,
+    convert_float_to_uint16,
+)
+
+import paddle
+import paddle.nn.functional as F
+from paddle.fluid import core
+
+paddle.enable_static()
+
+np.random.seed(123)
+paddle.seed(123)
+
+_set_use_system_allocator(True)
+
+
+def batch_norm_wrapper(
+    x,
+    running_mean,
+    running_variance,
+    weight,
+    bias,
+    is_test,
+    momentum,
+    epsilon,
+    data_format,
+    use_global_stats,
+):
+    # breakpoint()
+    y = F.batch_norm(
+        x,
+        running_mean,
+        running_variance,
+        weight,
+        bias,
+        training=not is_test,
+        momentum=momentum,
+        epsilon=epsilon,
+        data_format=data_format,
+        use_global_stats=use_global_stats,
+    )
+    z = F.relu(y)
+    return z
+
+
+class TestBatchNormOp(OpTest):
+    def setUp(self):
+        self.python_api = batch_norm_wrapper
+        self.public_python_api = batch_norm_wrapper
+        self.op_type = "batch_norm"
+        self.prim_op_type = "comp"
+        self.python_out_sig = ["Y"]
+        self.initConfig()
+        self.initTestCase()
+
+    def test_check_output(self):
+        if self.dtype not in ("uint16", "float16"):
+            self.check_output_with_place(
+                core.CPUPlace(),
+                no_check_set=None,
+                atol=self.fw_comp_atol,
+                rtol=self.fw_comp_rtol,
+                check_prim=True,
+                only_check_prim=True,
+            )
+        if paddle.is_compiled_with_cuda():
+            self.check_output_with_place(
+                core.CUDAPlace(0),
+                no_check_set=None,
+                atol=self.fw_comp_atol,
+                rtol=self.fw_comp_rtol,
+                check_prim=True,
+                only_check_prim=True,
+            )
+
+    def test_check_grad_x(self):
+        if self.dtype not in ("uint16", "float16"):
+            self.check_grad_with_place(
+                core.CPUPlace(),
+                ["X"],
+                ['Y'],
+                user_defined_grad_outputs=self.out_grad,
+                check_prim=True,
+                only_check_prim=True,
+            )
+        elif self.data_format == "NCHW" and paddle.is_compiled_with_cuda():
+            # origin batch_norm cuda kernel differ in nhwc x_grad whether to calculate scale_grad and bias_grad
+            self.check_grad_with_place(
+                core.CUDAPlace(0),
+                ["X"],
+                ['Y'],
+                user_defined_grad_outputs=self.out_grad,
+                check_prim=True,
+                only_check_prim=True,
+            )
+
+    def test_check_grad_scale_bias(self):
+        self.enable_cinn = False
+        self.rev_comp_atol = 1e-3
+        self.rev_comp_rtol = 1e-3
```

Comment:
why?

Reply:
origin kernel's grad of scale, bias differ with numpy under 1e-3